### PR TITLE
Implement delete task feature for TodoList

### LINKED ISSUE #165: TodoList: Implement delete task feature

### PR DESCRIPTION
**Pull request by michael-setia**
Fixes #165 
- Added implementation to delete a task from the todo list

### CODE REVIEW

File: src/main/java/tp/cap5buddy/logic/commands/DeleteTaskCommand.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package tp.cap5buddy.logic.commands;
+
+import tp.cap5buddy.contacts.ContactList;
+import tp.cap5buddy.modules.ModuleList;
+import tp.cap5buddy.todolist.Task;
+import tp.cap5buddy.todolist.TodoList;
+
```
Comment (jonasngs) [inline src/main/java/tp/cap5buddy/logic/commands/DeleteTaskCommand.java:7]: Perhaps javadocs could be added for the class here?

File: src/main/java/tp/cap5buddy/todolist/Criterion.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package tp.cap5buddy.todolist;
+
```
Comment (jonasngs) [inline src/main/java/tp/cap5buddy/todolist/Criterion.java:2]: Perhaps javadoc comments could be added for this enumeration to further explain the role of this enum class?

File: src/main/java/tp/cap5buddy/todolist/ParserUtilTodoList.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
+package tp.cap5buddy.todolist;
```
Comment (jonasngs) [inline src/main/java/tp/cap5buddy/todolist/ParserUtilTodoList.java:1]: We could use a common ParserUtil class to implement parsing methods which can be used across ModuleList, ContactList and TodoList in the future. For now, this is a sound implementation.

**Review by jonasngs — APPROVED** [review]
LGTM, albeit some minor issues with regards to missing javadocs.

File: src/main/java/tp/cap5buddy/todolist/ParserUtilTodoList.java (hunk 3 above)
Comment (michael-setia) [reply src/main/java/tp/cap5buddy/todolist/ParserUtilTodoList.java]: Yes, I will migrate all the methods to the ParserUtil class.

File: src/main/java/tp/cap5buddy/logic/commands/DeleteTaskCommand.java (hunk 1 above)
Comment (michael-setia) [reply src/main/java/tp/cap5buddy/logic/commands/DeleteTaskCommand.java:7]: Yes, I will add all the missing JavaDocs (I only added the javadocs that the CI requires for this PR)

**Pull request merged**